application/backend/src/db/schema.py

```diff
@@ -59,6 +59,7 @@ class JobDB(Base):
     id: Mapped[str] = mapped_column(primary_key=True, default=lambda: str(uuid4()))
     project_id: Mapped[str] = mapped_column(ForeignKey("projects.id"))
     type: Mapped[str] = mapped_column(String(64), nullable=False)
+    stage: Mapped[str] = mapped_column(String(64), nullable=False)  # training, validation, test
```

Comment:
I see it's also `idle` when the job hasn't started yet. Perhaps we can make this an enum? I.e. do something like `stage: Mapped[JobStage]` where `JobStage = 'idle' | 'training' | 'validation' | 'test'.